src/view.cpp

```diff
@@ -904,7 +958,13 @@ bool View::get_prev_feed(ItemListFormAction& itemlist)
 {
 	unsigned int feedpos;
 	if (feedlist_form->jump_to_previous_feed(feedpos)) {
-		prepare_query_feed(feedlist_form->get_feed());
+		try {
+			prepare_query_feed(feedlist_form->get_feed());
```

Comment:
Looks like this `prepare_query_feed()` call, together with the exact same exception handling, is repeated quite a few times.
Would it be possible to move the exception handling into `prepare_query_feed`, return a bool from it, and change these lines to the following?
```
if (!prepare_query_feed(feedlist_form->get_feed())) {
	return false;
}
```

Reply:
Yeah, I was debating this with myself too. On one hand, DRY requires to put all this error handling into `prepare_query_feed()`. On the other hand, `prepare_query_feed()` shouldn't even be a part of `View`, since query feed have nothing to do with views. Instead, it should be a method on `RssFeed` which returns an error, which is then handled in the calling code. In the end, I decided to do it that way so we don't have to refactor it later.

Does this sound acceptable, or am I just being lazy here? :)

Reply:
Sounds good